src/components/RenameHeader.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { FC, ReactNode, useState } from "react";
+import { Button, Input, Tooltip } from "@canonical/react-components";
+import SubmitButton from "components/SubmitButton";
+
+interface Props {
+  name: string;
+  parentItem: ReactNode;
+  centerControls?: ReactNode;
+  controls?: ReactNode;
+  isLoaded: boolean;
+  renameDisabledReason?: string;
+  onRename: (newName: string) => Promise<void>;
+  isRenaming: boolean;
+}
+
+const RenameHeader: FC<Props> = ({
+  name,
+  parentItem,
+  centerControls,
+  controls,
+  isLoaded,
+  onRename,
+  renameDisabledReason,
+  isRenaming,
+}) => {
+  const [newName, setNewName] = useState(name);
+  const [isRename, setRename] = useState(false);
+  const canRename = renameDisabledReason === undefined;
+
+  const handleRename = () => {
+    onRename(newName)
+      .then(() => setRename(false))
+      .catch(console.log);
+  };
+
+  return (
+    <div className="p-panel__header rename-header">
+      <h1 className="u-off-screen">{name}</h1>
+      {isLoaded ? (
+        <div className="p-panel__title">
+          <nav
+            key="breadcrumbs"
+            className="p-breadcrumbs"
+            aria-label="Breadcrumbs"
+          >
+            <ol className="p-breadcrumbs__items">
+              <li className="p-breadcrumbs__item">{parentItem}</li>
+              {isRename ? (
+                <li className="p-breadcrumbs__item rename">
+                  <Input
+                    autoFocus
+                    className="u-no-margin--bottom name-input"
+                    onChange={(e) => setNewName(e.target.value)}
+                    onKeyUp={(e) => e.key === "Enter" && handleRename()}
+                    type="text"
+                    value={newName}
+                  />
```

Comment:
I wonder if we can somehow plug in the `checkDuplicate` logic here. Because we're currently missing it for the rename, so we fall back to the default "postponed feedback".
One idea that I had is to wrap the `checkDuplicate` logic inside a function in the parent, and then pass it to this FC as a prop. Then you can call it inside `onChange` or `onKeyDown`.

Reply:
I think for validation it is better to use formik, so I rewrote the bit to have its own form with schema etc.
Also deduplicating now.